api/controllers/v1/jobs.ts

```diff
@@ -385,3 +412,100 @@ async function SubmitArchiveJob(jobId: string) {
   consoleLogger.info('submit archive job', JSON.stringify({ jobId: jobId, batchJobId: response.jobId }));
   await client.close();
 }
+
+/**
+ * Checks the signature payload as a rough validation that the request was made by
+ * the Snooty frontend.
+ * @param payload - stringified JSON payload
+ * @param signature - the Snooty signature included in the header
+ */
+function validateSnootyPayload(payload: string, signature: string) {
+  const secret = c.get<string>('snootySecret');
+  const expectedSignature = crypto.createHmac('sha256', secret).update(payload).digest('hex');
+  return signature === expectedSignature;
+}
+
+/**
+ * Performs post-build operations such as notifications and db updates for job ID
+ * provided in its payload. This is typically expected to only be called by
+ * Snooty's Gatsby Cloud source plugin.
+ * @param event
+ * @returns
+ */
+export async function SnootyBuildComplete(event: APIGatewayEvent): Promise<APIGatewayProxyResult> {
+  const consoleLogger = new ConsoleLogger();
+
+  if (!event.body) {
+    const err = 'SnootyBuildComplete does not have a body in event payload';
+    consoleLogger.error('SnootyBuildCompleteError', err);
+    return {
+      statusCode: 400,
+      headers: { 'Content-Type': 'text/plain' },
+      body: err,
+    };
+  }
+
+  // Keep lowercase in case header is automatically converted to lowercase
+  // The Snooty frontend should be mindful of using a lowercase header
+  const snootySignature = event.headers['x-snooty-signature'];
+  if (!snootySignature) {
+    const err = 'SnootyBuildComplete does not have a signature in event payload';
+    consoleLogger.error('SnootyBuildCompleteError', err);
+    return {
+      statusCode: 400,
+      headers: { 'Content-Type': 'text/plain' },
+      body: err,
+    };
+  }
+
+  if (!validateSnootyPayload(event.body, snootySignature)) {
+    const errMsg = 'Payload signature is incorrect';
+    consoleLogger.error('SnootyBuildCompleteError', errMsg);
+    return {
+      statusCode: 401,
+      headers: { 'Content-Type': 'text/plain' },
+      body: errMsg,
+    };
+  }
+
+  const payload: SnootyPayload = JSON.parse(event.body);
+  const { jobId } = payload;
+
+  if (!jobId) {
+    const errMsg = 'Payload missing job ID';
+    consoleLogger.error('SnootyBuildCompleteError', errMsg);
+    return {
+      statusCode: 401,
```

Comment:
NIT: I believe the status code should be a `400` if the payload is missing the `jobId` property